extensions/amp-addthis/0.1/amp-addthis.js

```diff
@@ -0,0 +1,361 @@
+const clickMonitor = new ClickMonitor();
+const activeToolsMonitor = new ActiveToolsMonitor();
+
+// `shouldRegisterView` is a shared flag that should be true only for the first built element on the
+// page, to prevent registering more than one view per page.
+let shouldRegisterView = true;
+
+// Redirection to prevent eslint issues.
+export function getConfigManager() {
+  return configManager;
+}
+
+class AmpAddThis extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?HTMLIFrameElement} */
+    this.iframe_ = null;
+
+    /** @private {string} */
+    this.pubId_ = '';
+
+    /** @private {string} */
+    this.widgetId_ = '';
+
+    /** @private {string} */
+    this.canonicalUrl_ = '';
+
+    /** @private {string} */
+    this.canonicalTitle_ = '';
+
+    /** @private {string} */
+    this.referrer_ = '';
+
+    /** @private {(?JsonObject<string, string>|null)} */
+    this.shareConfig_ = null;
+
+    /** @private {(?Object<string, string>|null)} */
+    this.atConfig_ = null;
+  }
+
+  /**
+   * @override
+   */
+  buildCallback() {
+    const pubId = this.element.getAttribute('data-pub-id');
+    const widgetId = this.element.getAttribute('data-widget-id');
+
+    // Required attributes
+    this.pubId_ = user().assert(
+        pubId,
+        'The data-pub-id attribute is required for <amp-addthis> %s',
+        this.element
+    );
+    this.widgetId_ = user().assert(
+        widgetId,
+        'The data-widget-id attribute is required for <amp-addthis> %s',
+        this.element
+    );
+
+    // Optional attributes
+    const ampDoc = this.getAmpDoc();
+    this.canonicalUrl_ = this.element.getAttribute('data-canonical-url') ||
+        ampDoc.getUrl();
+    this.canonicalTitle_ = this.element.getAttribute('data-canonical-title') ||
+        ampDoc.win.document.title;
+    this.shareConfig_ = this.getShareConfigAsJsonObject_();
+    this.atConfig_ = this.getATConfig_(ampDoc);
+
+    if (shouldRegisterView) {
+      // Register pageview (and setup analytics), only for the first
+      // amp-addthis element on the page
+      shouldRegisterView = false;
+
+      const viewer = Services.viewerForDoc(ampDoc);
+      const loc = parseUrl(this.canonicalUrl_);
+
+      viewer.whenFirstVisible()
+          .then(() => viewer.getReferrerUrl())
+          .then(referrer => {
+            this.referrer_ = referrer;
+
+            callLojson({
+              loc,
+              title: this.canonicalTitle_,
+              pubId: this.pubId_,
+              atConfig: this.atConfig_,
+              referrer,
+              ampDoc,
+            });
+
+            dwellMonitor.startForDoc(ampDoc);
+            scrollMonitor.startForDoc(ampDoc);
+            clickMonitor.startForDoc(ampDoc);
+          });
+
+      // Only the component that registers the page view listens for x-frame events.
+      this.setupListeners_({ampDoc, loc, pubId: this.pubId_});
```

Comment:
this can't be in the `buildCallback`, should move inside `whenFirstVisible` or `layoutcallback`

Reason is that this method sets up events listeners that end up triggering pixel drops or beacons (any communication that can cause IP leak in prerender mode is forbidden in AMP `buildCallabck` runs in prerender mode)